src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -28059,15 +28059,8 @@ void M(C c)
     }
     void Deconstruct(out C? x, out C? y) => throw null!;
 }";
-            // Note: we warn on the property initializer because it is a direct assignment to the backing field
```

Comment:
Hmm, this comment makes it feel like this was by-design. Since this initialization has to happen in internal code (the field initializer or constructor), perhaps it's expected that the user should have to use `!`

Reply:
The setter is marked `[AllowNull]` though.

---
In reply to: [349355581](https://github.com/dotnet/roslyn/pull/39939#discussion_r349355581) [](ancestors = 349355581)